Government refactor

LT-Rascek opened the pull request:
NB: Based on top of PR #71 as of 2022-02-01

Implement fixes and modifications to governments, realm laws, succession laws

### Governments
- [ ] Finish and have High Tribal Government Approved
   * Overhaul Realm Laws
   * Review Vassal Contracts
   * Review available succession laws
   * Review Government Character Modifiers
   * Implement Localization, including non-functional flag for character modifiers
   * Implement Game Concepts
- [ ] Finish and have Pre-Feudal Government Approved
   * Overhaul Realm Laws
   * Review Vassal Contracts
   * Review available succession laws
   * Review Government Character Modifiers
   * Implement Localization, including non-functional flag for character modifiers
   * Implement Game Concepts
- [ ] Finish and have Nomad Government Approved
   * Overhaul Realm Laws
   * Review Vassal Contracts
   * Review available succession laws
   * Review Government Character Modifiers
   * Implement Localization, including non-functional flag for character modifiers
   * Implement Game Concepts
- [ ] Finish and have Sub-Roman Government Approved
   * Overhaul Realm Laws
   * Review Vassal Contracts
   * Review available succession laws
   * Review Government Character Modifiers
   * Implement Localization, including non-functional flag for character modifiers
   * Implement Game Concepts
- [ ] Finish and have Eranshar Government Approved
   * Overhaul Realm Laws
   * Review Vassal Contracts
   * Review available succession laws
   * Review Government Character Modifiers
   * Implement Localization, including non-functional flag for character modifiers
   * Implement Game Concepts
- [ ] Finish and have Gupta Government Approved
   * Overhaul Realm Laws
   * Review Vassal Contracts
   * Review available succession laws
   * Review Government Character Modifiers
   * Implement Localization, including non-functional flag for character modifiers
   * Implement Game Concepts
- [ ] Finish and have Bureaucratic Government Approved
   * Overhaul Realm Laws
   * Review Vassal Contracts
   * Review available succession laws
   * Review Government Character Modifiers
   * Implement Localization, including non-functional flag for character modifiers
   * Implement Game Concepts

### Factions
- [x] Implement Tax Revolt Faction for Bureaucrats
- [ ] Overhaul casus belli & factions in light of having more than just crown_authority available

### On Actions
- [x] Implement a game_start on_action effect to properly set realm laws in a (mostly) automated way.
   * Better than doing it in history because it prevent the risk of overwriting during automatic assignment of realm laws
- [x] Rework succession laws to use game triggers instead of in history
   * More readily throws errors to error.log
   * Puts succession information in a central location instead of spread across hundreds of files

### Succession Laws
* NB: This should only be done _after_ the government designs are settled
- [ ] Overhaul Imperial Elective Succession Law (for Bureaucratic)
- [ ] Overhaul Mahestan Succession Law (for Eranshar)
- [ ] Implement Gubernatorial Succession Law (for Gubernatorials)
- [ ] Implement something like "elective gavelkind" for tribals
- [ ] Implement Turkish Succession for Turks

### Other
- [x] Add new government opinion modifiers (if possible)
   * ~~e.g., subroman_government_vassal_opinion = -10~~
   * Not possible it turns out

### Decision
- [x] Review Tribal to High Tribal decision
- [x] Impelement nomad settlement decision
- [ ] Review adopt subroman decision
- [ ] Review adopt bureaucratic decision
- [ ] Review adopt feudalism decision

### Tributaries and Foederati
- [x] Fixup Foederatus contract
- [x] Nix other vassal contracts for tributaries (so that only the Tributary contract applies); otherwise, we get very weird (low) values for levies/taxes.
   * Not possible; will need to think of better way around
- [x] Reimplement Tributary States
- [ ] Fix Tributaries, Foederati, Tributary States not being releasable by modify feudal contract window
- [ ] Review Tributaries, Foederati vis-a-vis councils
- [ ] Implement Tributary Wars
- [ ] Implement "Demand Tributary" Interaction

### Localization
- [ ] Import Eranshar localizations from WtWSMS

### Bugfixes
- [x] Fix independence on modifying High Tribal contracts
- [x] Zeno not inheriting realm taxation laws correctly and becoming subroman
- [x] Fix High Tribals not able to change succession laws
- [x] Fix "Change succession Law" Button not allowing non-clan/feudals to change succession law (error lists crown authority)
- [ ] Fix Title on action not properly reverting eranshar government on succession
- [ ] Fix interaction_modify_vassal_window.gui yield false "becomes independent" signals in foederatus negotiations

## Changes to Default Governments:
* Tribal Government
   * -50% Development Growth regardless of if the steward is working or not

## New Government Summary:
* High Tribal Government
   * Uses High Tribal Organization
      * Disparate Tribes: +10 direct vassal opinion, -40% Development Growth 
      * Low Organization: -10 direct vassal opinion, -30% Development Growth, Titles can be Revoked
      * Medium Organization: -20 direct vassal opinion, -20% Development Growth, Vassals can be Retracted
      * High Organization: -30 direct vassal opinion, -10% Development Growth, Internal Wars banned
   * Vassal Contract
         * Taxes and Levies depend on the rulers prestige (Tribal-like)
   * Feudal-type Features:
      * Development has same effects as feudalism (improves supplies, taxes, levies)
      * Regiments cost gold
      * Always has access to imprisonment
   * Tribal-type features:
      * -25% Title Creation Cost
      * -25% Army maintenance
      * Must use Partition OR Confederate Partition
      * Has access to Tribal Subjugation CB
   * Other Features:
      * Can only have up to Late Migration Era Innovations
      
* Pre-Feudal Government
   * Uses Pre-Feudal Authority
      * Autonomous Vassals: +10 direct vassal opinion, -20% Vassal Levy, -20% Vassal Tax
      * Tenuous Pre-Feudal Authority: +5 direct vassal opinion, -10% Vassal Levy, -10% Vassal Tax, +5% Development Growth, Titles can be revoked
      * Moderate Pre-Feudal Authority: -5 direct vassal opinion, +10% Development Growth, Vassals can be Retracted, Rulers can pick from Partition Succession
      * High Pre-Feudal Authority: -15 direct vassal opinion, +10% Vassal Levy, +10% Vassal Tax, +15% Development Growth, Rulers can pick from any valid Succession Law, Vassals can't wage internal wars
   * Vassal Contract
      * Pre-Feudal Obligations:
         * Exempt: 0% Tax, 0% Levies, +5 Opinion
         * Low: 2.5% Tax, 5% Levies
         * Medium: 5% Tax, 10% Levies, -5 Opinion
         * High: 10% Tax, 20% Levies, -10 Opinion
      * Religious Rights, War Declaration, Council Rights, and Title Revocation Contracts available
   * Other Features:
      * -15% Development Growth
      * Can only have up to Late Migration Era Innovations
      * Regiments cost gold
      * Always has access to imprisonment

* Nomad Government
   * Uses Tribal Authority
   * Vassal Contract
      * Nomadic Obligations
         * Alias of Clan Obligations
   * Other Features:
      * -50% Development Growth
      * Can only have up to Tribal Era Innovations
      * Regiments cost prestige
      * +25% Levies, -25% Taxes

## Notes:
* Title revocation revokes the whole duchy/kingdom in CK3, so no free title revocation given for gubernatorial vassals at this time.

## Other Changes:
* Religions
   * Zoroastrianism
      * Made male clerical gender; pretty sure female Zoroastrian priests are a recent innovation in Iran/the West (all resources I've read on the Sassanids make it pretty clear it was male-dominated)
      * Made Zoroastrian faiths have a temporal theocracy except Gayomarthism, Khurmmamsim (which seem Islam-related anyway)
         * Otherwise, priestly caste isn't really distinct when it actually was distinct and powerful
   * Graphics
      * Updated special_doctrine_zoroastrian_approved_cult, tenet_monument_building, tenet_zoroastrian_syncretism images
      * Added special_doctrine_zoroastrian_approved_infidel doctrine for Nestorian Church
         * Makes special_doctrine_zoroastrian_approved_cult Hostile instead of Evil
* On Actions
   * Create on_game_start action to make prefeudal vassals of Eranshar Govts be religiously protected
      * Reason: Prevents Persia from converting infidel titles in Asoristan wholesale

## Changes (ported from PR #71)
* Bureaucratic Changes:
   * Tweaked vote strength to make gubernatorial vassals have more sway given their diplomacy/stewardship/intrigue
      * Reason: gives players a reason to become gubernatorial vassals even when it costs them a greater degree of independence; specific weights should be reevaluated in testing
   * Overhauled Imperial elective vote weights in general in line with Succession Expanded data
* Added additional, non-functional flags to govts to provide users with additional information about govts

LT-Rascek commented:
Default Governments:
Tribal Government changes:
* -50% development growth rate instead of just when the steward is working

LT-Rascek commented:
## High Tribal Government Discussion
Reading through the dev diary from CK2: 

> It is a form of transition government, since it belongs to the same government category as Feudal or Bureaucratic and loses the disadvantages of being a tribe, but doesn’t get all the benefits of a Feudal ruler. With a small vassal limit and a small demesne any ruler that wants to build a larger empire will want to get rid of this government type.

And a quick summary from CK2:
* Retinues cost gold
* High Tribal Organization functions similar to crown authority in terms of opening options, but the council starts empowered
* Has all succession types opened

The last one bothered me, especially in terms of things like Frankish succession, so I made it locked to confederate partition. But the remainder of the choices should somewhat reflect what was in CK2, in light of the CK3 changes.

## High Tribal Government Summary
   * Uses High Tribal Centralization
      * Disparate Tribes: +10 direct vassal opinion, -40% Development Growth 
      * Low Centralization: -10 direct vassal opinion, -30% Development Growth, Titles can be Revoked
      * Medium Centralization: -20 direct vassal opinion, -20% Development Growth, Vassals can be Retracted
      * High Centralization: -30 direct vassal opinion, -10% Development Growth, Internal Wars banned
   * Feudal-type Features:
      * Development has same effects as feudalism (improves supplies, taxes, levies)
      * Regiments cost gold
      * Always has access to imprisonment
   * Tribal-type features:
      * -25% Title Creation Cost
      * -25% Army maintenance
      * Taxes and Levies depend on the rulers prestige
      * Must use Confederate Partition
      * Has access to Tribal Subjugation CB
   * Other Features:
      * Can only have up to Late Migration Era Innovations

[1] - Source: https://forum.paradoxplaza.com/forum/threads/wtwsms-presentation-info-news-links.829142/page-12#post-25664228

LT-Rascek commented:
## Pre-Feudal Government Discussion
In CK2 it was a pretty straight clone of Feudal Government. I followed mostly the same pattern in CK3; ~~Crown~~ Pre-Feudal Authority is more restricted than in Feudal Governments, Obligations are merged into a single axis for both taxes and levies, giving fewer of each. The only thing that really stands out is that I gave it a -15% development malus, which like High Tribal, can be mitigated with increasing Pre-Feudal Authority. It's done in part to prevent counties from developing to quickly from 476 to 700.

## Pre-Feudal Government Summary
   * Uses Pre-Feudal Authority
      * Autonomous Vassals: +10 direct vassal opinion, -20% Vassal Levy, -20% Vassal Tax
      * Tenuous Pre-Feudal Authority: +5 direct vassal opinion, -10% Vassal Levy, -10% Vassal Tax, +5% Development Growth, Titles can be revoked
      * Moderate Pre-Feudal Authority: -5 direct vassal opinion, +10% Development Growth, Vassals can be Retracted, Rulers can pick from Partition Succession
      * High Pre-Feudal Authority: -15 direct vassal opinion, +10% Vassal Levy, +10% Vassal Tax, +15% Development Growth, Rulers can pick from any valid Succession Law, Vassals can't wage internal wars
   * Vassal Contract
      * Pre-Feudal Obligations:
         * Exempt: 0% Tax, 0% Levies, +5 Opinion
         * Low: 2.5% Tax, 5% Levies
         * Medium: 5% Tax, 10% Levies, -5 Opinion
         * High: 10% Tax, 20% Levies, -10 Opinion
      * Religious Rights, War Declaration, Council Rights, and Title Revocation Contracts available
   * Other Features:
      * -15% Development Growth
      * Can only have up to Late Migration Era Innovations
      * Regiments cost gold
      * Always has access to imprisonment

LT-Rascek commented:
## Nomad Government Discussion
In CK2 nomads were, in a word, problematic. I've kept nomads to be mostly a tribal alias with a few exceptions. They use clan obligations instead of tribal obligations to tie into the Clan preference for Turks in vanilla. They also receive a bonus to levy size and men-at-arms cap, but a nerf to taxes. A decision was added to settle nomads to become tribals, similar to the convert to high tribal decision for tribals. Numbers and requirements for that decision will need a review after playtesting I'm sure.

## Pre-Feudal Government Summary
   * Uses Tribal Authority
   * Vassal Contract
      * Nomadic Obligations
         * Alias of Clan Obligations
   * Other Features:
      * -50% Development Growth
      * Can only have up to Tribal Era Innovations
      * Regiments cost prestige
      * +25% Levies, -25% Taxes

LT-Rascek commented:
## Pre-Feudal Government Discussion
In CK2 subroman decay into feudalism was driven by the event chain: `WtWSMS_govs.3 -> WtWSMS_govs.4 -> WtWSMS_govs.5` wherein failing to pay 2x-4x your yearly income (depending on where you were in the event chain) would lock you readily into becoming feudal. The irreversibility of the chain bothered me more than the price, since the event could come at extremely inopportune times. I prefer to not make such changes permanent and giving the players some ability to reverse course, so I've utilized representing this taxation infrastructure as a realm law instead. 

Unlike Crown Authority, Subroman Taxation costs both gold and prestige to improve; the gold cost increases with the realm size and costs more for every undeveloped county (undeveloped being <20 development in the current code base, but can be trivially tweaked).

Unlike other realm laws, "Subroman Taxation" does not allow you to voluntarily decrease it; events and the Liberty faction can, however, force you to drop it. A subroman ruler can only raise taxation every 50 years, while events fire about every 30 years or so to drive it down; it costs half what it would to raise taxation to avoid this event lowering your taxation. Given even then its pricey, there should be a steady degradation of taxation infrastructure for all but the most motivated rulers.

Voluntarily raising taxes yields negative opinion modifiers for both your owned counties and vassals for 25 years, emulating both the `maintained_roman_taxation_infrastructure` county modifier and the effects of rebuffing your vassal's attempt to persuade you to reduce the taxation infrastructure. The Liberty faction will also try to reduce the taxation infrastructure to tie in with the desire (in CK2) of your vassals wanting to do away with that infrastructure.

As the taxation declines, so to do various powers like seizing titles or banning vassal wars; this is done so that the lowest value of Roman Taxation has the same rights as the lowest value of Crown Authority, preventing a discontinuity in powers upon adopting Feudalism.

Overall, I think this provides more granularity and player interaction with the Feudalism transition than the CK2 event chain did, without moving away from the histography that was represented in CK2, as well as avoiding letting a bad RNG result (e.g.,heavily in debt fighting off barbarians) permanently force you to shift from feudalism (instead of, say, eventually reaching your goal of readopting bureaucracy). Furthermore the granularity helps show Subroman government as occupying the space between Bureaucratic/Gubernatorial and Feudal, much in the same way High Tribal occupies the space between Tribal and Feudal.

## Subroman Government Summary
   * Uses Subroman Taxation
      * Nonexistent Tax Infrastructure: -20% Domain Taxes, -20% Vassal Taxes
      * Decrepit Tax Infrastructure: -15% Domain Taxes, -15% Vassal Taxes, Titles can be revoked, Vassals can be retracted, +1 Domain Limit
      * Abandoned Tax Infrastructure: -10% Domain Taxes, -10% Vassal Taxes, Can pick between succession laws, +2 Domain Limit
      * Neglected Tax Infrastructure: -5% Domain Taxes, -5% Vassal Taxes, Vassals can't wage war between each other w/o a hook, +3 Domain Limit
      * Stable Tax Infrastructure: Can designate an heir, Vassal can't wage any war w/o a hook, +4 Domain Limit
   * Vassal Contract
      * Sub-Roman Taxes:
         * Exempt: 0% Tax, +10 Opinion
         * Low: +5% Taxes, +5 Opinion
         * Normal: +10% Taxes
         * High: +20% Taxes, -15 Opinion
         * Extortionate: +40% Taxes, -25 Opinion
      * Sub-Roman Levies:
         * Exempt: 0% Levies, +10 Opinion
         * Low: +10% Levies, +5 Opinion
         * Normal: +25% Levies
         * High: +35% Levies, -15 Opinion
         * Extortionate: +50% Levies, -25 Opinion
      * Religious Rights, War Declaration, Council Rights, and Title Revocation Contracts available
   * Other Features:
      * Create cadet branches
      * Can hold castles & cities
      * Domain Taxes +15%
      * Vassal Taxes +15%
      * Levies: -25%
      * Vassal Levies: -25%
      * Always has access to imprisonment

LT-Rascek commented:
## Bureaucratic Government Discussion
Bureaucrats have gone through several iterations. In CK2 they were only the rulers of some Empires with a number of changes (+200% Retinues, +5 Demesne size, free duchy revocation) but were very close to subroman. This didn't quite sit well with me, especially having most of the governments be subroman underneath the Emperor. Instead, I've looked more into _what_ we want to capture with bureacratic. As the CK2 flavor-text says:

>A bureaucratic administration relies on civil administrators, that can be appointed and dismissed at will, and less on landed vassals. Rulers can personally rule more territory and gain more taxes, and maintain more experienced professional soldiers at the cost of being able to raise less levies.

So I've reworked Bureaucrats a good bit to center them more around the concept of civil service. Under this model, bureaucrats utilize "Bureaucratic Taxation", which attempts to model the heavily-centralized taxation model that developed in the late Empire. Rather than a flat 15% domain tax benefit, the domain tax is set by the Emperor at any of five levels; unlike Crown Authority, we can go from level 1 to level 5 instantly (at a large cost of gold and prestige).

The imperial tax rate is set by the Emperor and is propagated down the de facto hierarchy to all vassals; individual bureaucrats have no control over the taxation rate. The vassal contract controls how much in taxes the vassal pays; 0-50% in 10% increments while the levy rate is fixed at 25%.

Unlike feudal contracts, all taxation does is change the domain tax rate. Bureaucrats can _always_ revoke titles, retract vassals, &c. This greatly strengthens a liege's powers against his vassals. On the flip side, bureaucrats are more likely to support rival claimants (same degree as say an incorrect gender ruler would have), and will have more power in succession.

The Imperial title uses Imperial Elective Succession (to be overhauled) while bureaucratic vassals use Gubernatorial Succession (to be implemented) on duchy and kingdom titles. I plan to give the Emperor the ability (via character modifiers) to influence elections of his vassals.

My hope is that these changes will better differentiate Bureaucratic from Sub-Roman and later feudal governments by giving the Emperor significant powers and being tax-focused, if being at greater risk of deposition by his vassals than Eranshar would.

As for gubernatorial government, it became far too clunky and difficult to use so I've streamlined it into bureaucratic government and greatly reduced code complexity as a result.

## Bureaucratic Government Summary
   * Uses Bureaucratic Taxation
      * Minimal Taxation, +10 Popular Opinion
      * Low Taxation:  +5% Domain Taxes, +5 Popular Opinion
      * Moderate Taxation: +10% Domain Taxes
      * High Taxation: +15% Domain Taxes, -10 Popular Opinion
      * Very High Taxation: +25% Domain Taxes, -20 Popular Opinion
   * Vassal Contract
      * Bureaucratic Budget:
         * Very Large Budget: 0% Taxes, 25% Levies
         * Large Budget: 10% Taxes, 25% Levies
         * Normal Budget: 20% Taxes, 25% Levies
         * Low Budget: 30% Taxes, 25% Levies
         * Non-Existent Budget: 50% Taxes, 25% Levies
      * Bureaucratic Special Contracts available (Exarchate, Theme)
   * Other Features:
      * Create cadet branches
      * Can hold castles & cities
      * Empires use Imperial Elective Succession, Duchy and Kingdom Titles use Gubernatorial Succession
      * Only valid if Emperor or have a Bureaucratic Liege
      * Titles can **always** be revoked
      * Vassals can **always** be retracted
      * Refusing revocation/retraction justifies imprisonment
      * Titles cannot be inherited outside the realm
      * Vassals cannot wage war without a hook on the liege
      * Rulers can designate an heir
      * Men-At-Arms Limit: +5
      * Domain Limit: +5
      * Direct Vassal Opinion: -10
      * Always has access to imprisonment

loup99 commented:
> ## High Tribal Government Discussion
> Reading through the dev diary from CK2:
> 
> > It is a form of transition government, since it belongs to the same government category as Feudal or Bureaucratic and loses the disadvantages of being a tribe, but doesn’t get all the benefits of a Feudal ruler. With a small vassal limit and a small demesne any ruler that wants to build a larger empire will want to get rid of this government type.
> 
> And a quick summary from CK2:
> 
> * Retinues cost gold
> * High Tribal Organization functions similar to crown authority in terms of council options
> * Has all succession types opened
> 
> The last one bothered me, especially in terms of things like Frankish/Germanic succession, so I made it locked to confederate partition; if there's information on other high tribal kingdoms utilizing different succession patterns, please let me know. But the remainder of the choices should somewhat reflect what was in CK2, in light of the CK3 changes. Instead of the flat -50% development growth nerf Tribal has, development growth is a function of the realm law, with increasing centralization improving the rulers ability to impose his will. Like other primitive governments, it won't have access to Medieval innovations, and it retains a number of Tribal-ish features, if somewhat toned down.
> 
> ## High Tribal Government Summary
> * Uses High Tribal Organization
>   
>   * Disparate Tribes: +10 direct vassal opinion, -40% Development Growth
>   * Low Organization: -10 direct vassal opinion, -30% Development Growth, Titles can be Revoked
>   * Medium Organization: -20 direct vassal opinion, -20% Development Growth, Vassals can be Retracted
>   * High Organization: -30 direct vassal opinion, -10% Development Growth, Internal Wars banned
> * Feudal-type Features:
>   
>   * Development has same effects as feudalism (improves supplies, taxes, levies)
>   * Regiments cost gold
>   * Always has access to imprisonment
> * Tribal-type features:
>   
>   * -25% Title Creation Cost
>   * -25% Army maintenance
>   * Taxes and Levies depend on the rulers prestige
>   * Must use Confederate Partition
>   * Has access to Tribal Subjugation CB
> * Other Features:
>   
>   * Can only have up to Late Migration Era Innovations
> 
> [1] - Source: https://forum.paradoxplaza.com/forum/threads/wtwsms-presentation-info-news-links.829142/page-12#post-25664228

On succession, it was the major difference between tribal and high tribal. While tribal kingdoms would shatter upon succession without a charismatic and strong leader, high tribal ones remained united no matter the heir. This shatter mechanic meant the main heir in most cases did not retain the primary title if it was a de jure kingdom.

loup99 commented:
> ## Sub-Roman Government Discussion
> In CK2 subroman decay into feudalism was driven by the event chain: `WtWSMS_govs.3 -> WtWSMS_govs.4 -> WtWSMS_govs.5` wherein failing to pay 2x-4x your yearly income (depending on where you were in the event chain) would lock you readily into becoming feudal. The irreversibility of the chain bothered me more than the price, since the event could come at extremely inopportune times. I prefer to not make such changes permanent and giving the players some ability to reverse course, so I've utilized representing this taxation infrastructure as a realm law instead.
> 
> Unlike Crown Authority, Subroman Taxation costs both gold and prestige to improve; the gold cost increases with the realm size and costs more for every undeveloped county (undeveloped being <20 development in the current code base, but can be trivially tweaked).
> 
> Unlike other realm laws, "Subroman Taxation" does not allow you to voluntarily decrease it; events and the Liberty faction can, however, force you to drop it. A subroman ruler can only raise taxation every 50 years, while events fire about every 30 years or so to drive it down; it costs half what it would to raise taxation to avoid this event lowering your taxation. Given even then its pricey, there should be a steady degradation of taxation infrastructure for all but the most motivated rulers.
> 
> Voluntarily raising taxes yields negative opinion modifiers for both your owned counties and vassals for 25 years, emulating both the `maintained_roman_taxation_infrastructure` county modifier and the effects of rebuffing your vassal's attempt to persuade you to reduce the taxation infrastructure. The Liberty faction will also try to reduce the taxation infrastructure to tie in with the desire (in CK2) of your vassals wanting to do away with that infrastructure.
> 
> As the taxation declines, so to do various powers like seizing titles or banning vassal wars; this is done so that the lowest value of Roman Taxation has the same rights as the lowest value of Crown Authority, preventing a discontinuity in powers upon adopting Feudalism.
> 
> Overall, I think this provides more granularity and player interaction with the Feudalism transition than the CK2 event chain did, without moving away from the histography that was represented in CK2, as well as avoiding letting a bad RNG result (e.g.,heavily in debt fighting off barbarians) permanently force you to shift from feudalism (instead of, say, eventually reaching your goal of readopting bureaucracy). Furthermore the granularity helps show Subroman government as occupying the space between Bureaucratic/Gubernatorial and Feudal, much in the same way High Tribal occupies the space between Tribal and Feudal.
> 
> ## Subroman Government Summary
> * Uses Subroman Taxation
>   
>   * Nonexistent Tax Infrastructure: -20% Domain Taxes, -20% Vassal Taxes
>   * Decrepit Tax Infrastructure: -15% Domain Taxes, -15% Vassal Taxes, Titles can be revoked, Vassals can be retracted, +1 Domain Limit
>   * Abandoned Tax Infrastructure: -10% Domain Taxes, -10% Vassal Taxes, Can pick between succession laws, +2 Domain Limit
>   * Neglected Tax Infrastructure: -5% Domain Taxes, -5% Vassal Taxes, Vassals can't wage war between each other w/o a hook, +3 Domain Limit
>   * Stable Tax Infrastructure: Can designate an heir, Vassal can't wage any war w/o a hook, +4 Domain Limit
> * Vassal Contract
>   
>   * Sub-Roman Taxes:
>     
>     * Exempt: 0% Tax, +10 Opinion
>     * Low: +5% Taxes, +5 Opinion
>     * Normal: +10% Taxes
>     * High: +20% Taxes, -15 Opinion
>     * Extortionate: +40% Taxes, -25 Opinion
>   * Sub-Roman Levies:
>     
>     * Exempt: 0% Levies, +10 Opinion
>     * Low: +10% Levies, +5 Opinion
>     * Normal: +25% Levies
>     * High: +35% Levies, -15 Opinion
>     * Extortionate: +50% Levies, -25 Opinion
>   * Religious Rights, War Declaration, Council Rights, and Title Revocation Contracts available
> * Other Features:
>   
>   * Create cadet branches
>   * Can hold castles & cities
>   * Domain Taxes +15%
>   * Vassal Taxes +15%
>   * Levies: -25%
>   * Vassal Levies: -25%
>   * Always has access to imprisonment

As mentioned in the conversation I disagree with the portrayal of the CK2 system. The redesign is based upon a misunderstanding of how the original event chain worked. The explanation about supposed RNG illustrates this, the multi-factorial system that existed in CK2 is turned into a one-factor transition centred only around the taxation infrastructure. When your realm is devastated by war, that should have concrete consequences on how the government functions. When your realm becomes increasingly militarised then that modifies how the government works. I do not agree with the removal of the militarisation and warfare mechanics for Sub-Roman rulers, which complemented the taxation infrastructure decay.

loup99 commented:
> ## Bureaucratic Government Discussion
> Bureaucrats have gone through several iterations. In CK2 they were only the rulers of some Empires with a number of changes (+200% Retinues, +5 Demesne size, free duchy revocation) but were very close to subroman. This didn't quite sit well with me, especially having most of the governments be subroman underneath the Emperor. Instead, I've looked more into _what_ we want to capture with bureacratic. As the CK2 flavor-text says:
> 
> > A bureaucratic administration relies on civil administrators, that can be appointed and dismissed at will, and less on landed vassals. Rulers can personally rule more territory and gain more taxes, and maintain more experienced professional soldiers at the cost of being able to raise less levies.
> 
> So I've reworked Bureaucrats a good bit to center them more around the concept of civil service. Under this model, bureaucrats utilize "Bureaucratic Taxation", which attempts to model the heavily-centralized taxation model that developed in the late Empire. Rather than a flat 15% domain tax benefit, the domain tax is set by the Emperor at any of five levels; unlike Crown Authority, we can go from level 1 to level 5 instantly (at a large cost of gold and prestige).
> 
> The imperial tax rate is set by the Emperor and is propagated down the de facto hierarchy to all vassals; individual bureaucrats have no control over the taxation rate. The vassal contract controls how much in taxes the vassal pays; 0-50% in 10% increments while the levy rate is fixed at 25%.
> 
> Unlike feudal contracts, all taxation does is change the domain tax rate. Bureaucrats can _always_ revoke titles, retract vassals, &c. This greatly strengthens a liege's powers against his vassals. On the flip side, bureaucrats are more likely to support rival claimants (same degree as say an incorrect gender ruler would have), and will have more power in succession.
> 
> The liberty faction does not exist for Bureaucrats (since all taxation is centrally controlled, and aristocrats always want more taxes). Instead, I'm (in the process) of writing a variant of the peasant faction that, if it's revolt succeeds, will force the tax rate to decline, modeled on historic Roman tax revolts like the Bucolic War, Bagaudae, Basil the Copper Hand, &c.
> 
> The Imperial title uses Imperial Elective Succession (to be overhauled) while bureaucratic vassals use Gubernatorial Succession (to be implemented) on duchy and kingdom titles. I plan to give the Emperor the ability (via character modifiers) to influence elections of his vassals.
> 
> My hope is that these changes will better differentiate Bureaucratic from Sub-Roman and later feudal governments by giving the Emperor significant powers and being tax-focused, if being at greater risk of deposition by his vassals than Eranshar would. At the very least, the architecture has been laid here for a government with far more centralized control than feudal could hope to have.
> 
> As for gubernatorial government, it became far too clunky and difficult to use so I've streamlined it into bureaucratic government and greatly reduced code complexity as a result.
> 
> ## Bureaucratic Government Summary
> * Uses Bureaucratic Taxation
>   
>   * Minimal Taxation, +10 Popular Opinion
>   * Low Taxation:  +5% Domain Taxes, +5 Popular Opinion
>   * Moderate Taxation: +10% Domain Taxes
>   * High Taxation: +15% Domain Taxes, -10 Popular Opinion
>   * Very High Taxation: +25% Domain Taxes, -20 Popular Opinion
> * Vassal Contract
>   
>   * Bureaucratic Budget:
>     
>     * Very Large Budget: 0% Taxes, 25% Levies
>     * Large Budget: 10% Taxes, 25% Levies
>     * Normal Budget: 20% Taxes, 25% Levies
>     * Low Budget: 30% Taxes, 25% Levies
>     * Non-Existent Budget: 50% Taxes, 25% Levies
>   * Bureaucratic Special Contracts available (Exarchate, Theme)
> * Other Features:
>   
>   * Create cadet branches
>   * Can hold castles & cities
>   * Empires use Imperial Elective Succession, Duchy and Kingdom Titles use Gubernatorial Succession
>   * Only valid if Emperor or have a Bureaucratic Liege
>   * Titles can **always** be revoked
>   * Vassals can **always** be retracted
>   * Refusing revocation/retraction justifies imprisonment
>   * Titles cannot be inherited outside the realm
>   * Vassals cannot wage war without a hook on the liege
>   * Rulers can designate an heir
>   * Men-At-Arms Limit: +5
>   * Domain Limit: +5
>   * Direct Vassal Opinion: -10
>   * Always has access to imprisonment

I think this is more meaningful than the unfinished mechanics we had in CK2, but I don't understand how the vassals would be bureaucratic instead of gubernatorial/sub-roman.

LT-Rascek commented:
> On succession, it was the major difference between tribal and high tribal. While tribal kingdoms would shatter upon succession without a charismatic and strong leader, high tribal ones remained united no matter the heir. This shatter mechanic meant the main heir in most cases did not retain the primary title if it was a de jure kingdom.

Translating that into CK3's partition system, high tribals should probably then (at least start) with partition (no new titles created, but still same partition logic). Perhaps the Merovingians should be locked into Confederate Partition regardless; either way, it's a (mostly) simple fix.

Only minor hiccup is that Partition succession is only accessible with the early medieval innovation "Hereditary rule". Shifting it to, say, writing innovation is not that big of an issue though; just thought I'd note.

> I think this is more meaningful than the unfinished mechanics we had in CK2, but I don't understand how the vassals would be bureaucratic instead of gubernatorial/sub-roman. Having Sub Roman vassals without taxation makes more sense to me.

A **big part** of it is merely mechanical; all having the same government means I don't have to add some convoluted logic to manage the title hand-offs because of the implication of `can_get_government` will lead to situations where fallback government gets invoked. (e.g., if I'm Eranshar, should I keep the bureaucratic government? What if I'm a tribal? What if I'm this religion or that religion, &c.) Which was the issue I was having with gubernatorial leading me to jettison it, because it worked _much_ nicer as a singular government; Zeno no longer became subroman and led to a giant swath of feudal lords after 3 days. 

The other advantage is a single set of realm laws for both liege and all vassals that is very simply centrally controlled; the liege controls the tax rate by propagating that down the hierarchy via `on_pass` while the line `is_independent_ruler = yes` in `can_pass` means only the liege controls the tax rate and the domain tax rate bonus is a function of something fairly centrally set. There's no duplicate set of laws to deal with, although clever writing of the `can_keep` code could make that not as big of deal, even for adding other governments to it, now that I think about it.

But, now that I've said it out loud, I can probably jettison `can_get_government` _in toto_ and avoid all the headaches I've had to work around with a few simple fixes in a few simple places that are _already_ mostly in place... While "make everyone a bureaucrat" definitely reduces the amount of code and the amount of headaches we'd run into, I actually think I can sidestep the issues I was having [1]...

So I'm going to think about it but I can probably bring gubernatorial governments back in without all the fall back government issues I was having earlier _and_ make the code base cleaner than it was prior to PR72.

=====

So here's the pros and cons of keeping bureaucratic from my view:
Pro: Everyone being bureaucratic keeps the code simpler; the fewer the governments, the less maintenance needs to be done, the less custom triggers need to written and reviewed, &c.
Pro: Keeps various bits of code relatively siloed by government type instead of having them bleed into each other
Pro: It's more aesthetically pleasing (to me at least) to have it as a single block

![its_so_pretty](https://user-images.githubusercontent.com/55886300/151688174-4c5c780b-6efc-4097-b38e-32f80e8f52fa.jpg)

Con: Not having gubernatorial vassals isn't as thematic, which might confuse players
Con: Breaks with the sort of logic and dichotomy that was in CK2, which may impact events when imported

Think about it for a bit and let me know

-----
[1] Thanks for being a rubber duck!
(In case you've never heard the expression before:  <https://en.wikipedia.org/wiki/Rubber_duck_debugging>)

loup99 commented:
> > On succession, it was the major difference between tribal and high tribal. While tribal kingdoms would shatter upon succession without a charismatic and strong leader, high tribal ones remained united no matter the heir. This shatter mechanic meant the main heir in most cases did not retain the primary title if it was a de jure kingdom.
> 
> Translating that into CK3's partition system, high tribals should probably then (at least start) with partition (no new titles created, but still same partition logic). Perhaps the Merovingians should be locked into Confederate Partition regardless; either way, it's a (mostly) simple fix.
> 
> Only minor hiccup is that Partition succession is only accessible with the early medieval innovation "Hereditary rule". Shifting it to, say, writing innovation is not that big of an issue though; just thought I'd note.
> 
> > I think this is more meaningful than the unfinished mechanics we had in CK2, but I don't understand how the vassals would be bureaucratic instead of gubernatorial/sub-roman. Having Sub Roman vassals without taxation makes more sense to me.
> 
> A **big part** of it is merely mechanical; all having the same government means I don't have to add some convoluted logic to manage the title hand-offs because of the implication of `can_get_government` will lead to situations where fallback government gets invoked. Which was the issue I was having with gubernatorial leading me to jettison it, because it worked _much_ nicer as a singular government; Zeno no longer became subroman and led to a giant swath of feudal lords after 3 days.
> 
> The other advantage is a single set of realm laws for both liege and all vassals that is very simply centrally controlled; the liege controls the tax rate by propagating that down the hierarchy via `on_pass` while the line `is_independent_ruler = yes` in `can_pass` means only the liege controls the tax rate and the domain tax rate bonus is a function of something fairly centrally set. There's no duplicate set of laws to deal with, although clever writing of the `can_keep` code could make that not as big of deal, even for adding other governments to it, now that I think about it.
> 
> But, now that I've said it out loud, I can probably jettison `can_get_government` _in toto_ and avoid all the headaches I've had to work around with a few simple fixes in a few simple places that are _already_ mostly in place... While "make everyone a bureaucrat" definitely reduces the amount of code and the amount of headaches we'd run into, I actually think I can sidestep the issues I was having [1]...
> 
> So I'm going to think about it but I can probably bring gubernatorial governments back in without all the fall back government issues I was having earlier _and_ make the code base cleaner than it was prior to PR72.
> 
> =====
> 
> So here's the pros and cons of keeping bureaucratic from my view: Pro: Everyone being bureaucratic keeps the code simpler; the fewer the governments, the less maintenance needs to be done, the less custom triggers need to written and reviewed, &c. Pro: Keeps various bits of code relatively siloed by government type instead of having them bleed into each other Pro: It's more aesthetically pleasing (to me at least) to have it as a single block
> 
> ![its_so_pretty](https://user-images.githubusercontent.com/55886300/151688174-4c5c780b-6efc-4097-b38e-32f80e8f52fa.jpg)
> 
> Con: Not having gubernatorial vassals isn't as thematic, which might confuse players Con: Breaks with the sort of logic and dichotomy that was in CK2, which may impact events when imported
> 
> Think about it for a bit and let me know
> 
> [1] Thanks for being a rubber duck! (In case you've never heard the expression before: https://en.wikipedia.org/wiki/Rubber_duck_debugging)

I don't want to add a lot of work for you, but the question that comes to mind is if the solution of keeping the removal of the Gubernatorial government but making the vassals Sub-Roman wouldn't be easier to handle in case of independence, since we don't want the vassals to become independent Bureaucratic realms and turn Sub-Roman in that scenario.

LT-Rascek commented:
## Eranshar Government Discussion
For reference, my primary sources for Sasanid history are [1,2]

To refresh, here's a synopsis of the CK2 description of Eranshar
> 1). Can hold Castle & City holdings without penalties
> 2). Can hold Tribal holdings without penalties for counties with your culture
> 3). Can build Castle, City holdings
> 4). Can call Tribal & Eranshahr government vassals to arms instead of using liege levies
> 5). Rulers can raid infidel neighbors for loot
> 6). Other vassals will not object to vassal retraction
> 7). Can move capital within the same lifetime every 600 months
> 8). Tribal & Eranshahr government vassals are not included in vassal limit calculation
> 9). Can't grant kingdoms, empires to characters with a government from a different group

On top of that, obligations followed the same form that other forms of feudal-like government (shifting different 
vassal obligations to and from levies and taxes).

While some of these I'd agree with (e.g., 6,7,9), there's other's I'd object to:

1). The Sasanid government had a quadripartite notion of class in descending order: priests, warriors, farmers/husbandmen, and artisans/merchants. As such, allowing the grandees and major nobles to hold cities seems a bit odd. So I'd propose only letting Eranshar government hold castles and let republics hold cities like vanilla CK3 feudalism [1; Ch.2, Pesag: Class Division].
2). Some tensions existed between the nomads and urban centers, so allowing any ownership of tribal holdings seems problematic at best [1; Ch.2, Organization of the Society: Nomadism vs Urbanism].

So with that out of the way, some more discussion:

Sasanian history seems to focus on the intersection of three major groups:  the gentry/nobility, the priesthood, and the Shahanshah. Early on, the former two definitely dominated until at least the rise of Shapur II [2; Ch. 2.1.3]. In particular, the priesthood was central to the initial establishment of the Sasanian state, which is why the religious structure of Eranshar has some noticeable differences compared to that of Rome [1; Ch.1: Ardashir I and the Establishment of the Empire]. So whatever model we want to use should have some mechanics that help reflect these differing trends.

So two sets of realm laws should apply (at least for the Shahanshah:  one for the secular grandees and another for the clerical hierarchy. This should help distinguish it from the religious/government interactions that formed Roman governance.

For Secular Authority, it should represent the tension between the most powerful families in the realm and the Shah. The attempt at curtailing the power of the Parthian clans was the primary source of strife in the later part of the Sasanian dynasty [2; Ch.2]. In this sense, its more similar to Feudal Crown Authority than anything else; at low levels the king has significantly curtailed powers and increasing his authority increases his power over vassals and increases their base taxes/levies owed.

A key difference however, is that while titles are always revocable, it costs _**renown**_ to revoke titles. The cost varies depending on the target dynasty's renown and the Shah's authority, which will discourage unlanding powerful vassals. Furthermore, at low levels of Secular Authority, the Shahanshah must only grant titles to powerful families. This represents the 
 overall power the Parthian Clans had against the Shahanshah early in the empire, wherein they tended to give eastern titles to those from the powerful clans. Secular Authority should also play a role in the vote strength of the Eranshar vassals for Mahestan succession.

It's not all downsides; Eranshar rulers always consider Eranshar lieges to be rightful if the top liege owns e_persia and is Eranshar*. As for obligations, tying it to opinion like clan government seems warranted, with Secular Authority helping control the minimum value. 

* - NB: this will cause the levies/taxes of the vassals tab to look a little wonky, but it's how I got around the hard-coded "Not Rightful Liege" maluses.

Uses clan-type obligations (which dovetails with the confederate-type structure of the North & East of the Empire under Parthian clan dominion [2; Ch.2)], but with higher lows and more taxes (a more legalistic, centralized government). This ties in with the fact that, at least initially, the Eranshar structure was built upon the previous Arascid confederate structure.

Opinion | -100 |   -50 |   0 |    50 | 100
--------|------|-------|-----|-------|-----
Tax     | 10%  |   20% | 30% |   40% | 50%
Levy    | 10%  | 22.5% | 35% | 47.5% | 60%

The other primary difference between the Sasanians and Romans is the near-total fusion of the Zoroastrian priesthood with the state in a way that was fundementally differenent then what occured in Rome; the Zoroastrian priesthood was central to the Sasanian rise to power and fully integrated with the state from the beginning. As such, I've chosen to represent it with a second relam law titled "Clerical Influence" demonstrating the overall influence of the theocrats on the government. It's only available to the Emperor who has his HOF as a vassal.

It primarily changes clerical opinions, theocracy tax/levy rates, and different faith popular opinions but with two major exceptions:  As the clerics gain/loose power, the clerical appointment of Asvan faiths change (to show how the Shahanshah has more sway over the clerical hierarchy) and at higher levels, temporal condemnation is available (which sort-of occurred against some Sasanian officials historically; e.g., Yazdegerd I "the Sinner", Mehr-Narseh condemnation [3]).

One of the changes I think is warranted is changing Asvan Zoroastrian faiths to be theocratic instead of lay clergy as the Zoroastrian priesthood was really closer in function to the Christian or the Egyptian priesthoods in that they  were their own separate power base instead of fully under the secular rulers thumb.

Finally, given the historic (although it might be more a reflection of later views than fact [1; Ch.2, Pesag: Class Division]), republic vassals should offer a total 50% taxes and 25% levies at the cost of -20 opinion for Eranshar rulers, representing the fact that the merchant class was effective in its state industries even if it was heavily regulated and dismissed by the nobility.

Other changes I've added include:
* Giving Nestorianism the "Legalized Faith" to make them hostile towards (Asvan) Zoroastrians instead of evil
* Zoroastrians now consider the same family as evil (see Manicheans)
* Giving Khurmazta and Arewordi Asvan instead of being left in family
* Making Asvan treat each other as astray instead of righteous

Overall, I think this will help give Eranshar both a distinct government flavor from Roman Bureaucracy while also being an overall more accurate picture of Sasanian government

LT-Rascek commented:
## Gupta Government Discussion
To start off with, in CK2, Gupta was pretty close clone of feudal. I read through the planning discussion you listed [1] and implemented some additional features that should (somewhat) model the decentralized & scholarly bureaucrat dynamic. The overall model is still close to Vanilla CK3 feudalism, but with a few twists.

I was able to require titles be granted to family members (if it is actually possible) like Enlil wanted in the PDX forum post. Beyond that, Gupta's use centralization, which is very similar to crown authority but instead increases the **base** contribution instead of being a simple multiplier. Gupta obligations are likewise very similar to feudal if slightly more powerful. I removed some vassal contracts like religious rights (a non-issue really during the Gupta period) and council rights (because of a scholar-based bureaucracy; I couldn't change powerful vassals wanting a seat on the council. That's hardcoded).

Overall, pretty similar to the CK2 model with a few features that were desired but unimplementable in CK2

## Gupta Government Summary
   * Uses Centralization
      * Decentralized Realm
      * Low Centralization: Direct Vassal Opinion -10, Base Levy/Tax Contribution +10%, Titles can be revoked, Vassals can be retracted
      * Moderate Centralization: Direct Vassal Opinion -20, Base Levy/Tax Contribution +20%, Titles can be revoked, Vassals can be retracted, Vassals cannot wage war on other vassals without a hook on the liege, all vassals who refuse title revocation/vassal retraction are criminals
      * High Centralization: Direct Vassal Opinion -30, Base Levy/Tax Contribution +30%, vassals cannot wage any war without a hook on their liege
   * Vassal Contract
      * Gupta Taxes
         * None: +0%, +10 Opinion
         * Low: +5%, +5 Opinion
         * Normal: +10%
         * High: +15%, -15 Opinion
         * Extortionate: +20%, -35 Opinion
      * Gupta Levies
         * None: +0%, +10 Opinion
         * Low: +10%, +5 Opinion
         * Normal: +20%
         * High: +30%, -15 Opinion
         * Extortionate: +40%, -35 Opinion
      * Special Contracts: Succession Rights, War Declaration, Title Revocation
   * Other Features:
      * Can **not** create cadet branches
      * Titles can **always** be revoked
      * Vassals can **always** be retracted
      * Vassals must give titles to dynasty members if possible
      * Titles cannot be inherited outside the realm
      * Rulers can select any valid succession law for their **realm**
      * Rulers can designate an heir
      * Development Growth: +25%
      * Monthly Control Growth: +0.1
      * Always has access to imprisonment

[1] https://forum.paradoxplaza.com/forum/threads/the-other-empires-the-gupta.962892/
